From bd480c482c3aeed2c27831a7de641d53103cf77e Mon Sep 17 00:00:00 2001
From: Kim <elektroschmock78@googlemail.com>
Date: Thu, 12 Nov 2015 09:22:34 +0100
Subject: [PATCH 06/28] Unbreak QWERTZ keyboards

Adding "more" keys to ALL the keys broke QWERTZ keyboards.
Fix it by adding seperate XMLs for QWERTZ.

Change-Id: Iccf10139cf02cab2a773c7085aadac8538359a16
---
 java/res/xml-sw600dp/rows_qwertz.xml    |  2 +-
 java/res/xml/rowkeys_qwertz2.xml        | 30 +++++++++++++++++
 java/res/xml/rowkeys_qwertz2_left5.xml  | 44 +++++++++++++++++++++++++
 java/res/xml/rowkeys_qwertz2_right4.xml | 44 +++++++++++++++++++++++++
 java/res/xml/rowkeys_qwertz3.xml        | 20 +++++++++--
 java/res/xml/rows_qwertz.xml            |  2 +-
 6 files changed, 137 insertions(+), 5 deletions(-)
 create mode 100644 java/res/xml/rowkeys_qwertz2.xml
 create mode 100644 java/res/xml/rowkeys_qwertz2_left5.xml
 create mode 100644 java/res/xml/rowkeys_qwertz2_right4.xml

diff --git a/java/res/xml-sw600dp/rows_qwertz.xml b/java/res/xml-sw600dp/rows_qwertz.xml
index 3b59dec54..747dafdce 100644
--- a/java/res/xml-sw600dp/rows_qwertz.xml
+++ b/java/res/xml-sw600dp/rows_qwertz.xml
@@ -36,7 +36,7 @@
         latin:keyWidth="9.0%p"
     >
         <include
-            latin:keyboardLayout="@xml/rowkeys_qwerty2"
+            latin:keyboardLayout="@xml/rowkeys_qwertz2"
             latin:keyXPos="4.5%p" />
         <Key
             latin:keyStyle="enterKeyStyle"
diff --git a/java/res/xml/rowkeys_qwertz2.xml b/java/res/xml/rowkeys_qwertz2.xml
new file mode 100644
index 000000000..10878a019
--- /dev/null
+++ b/java/res/xml/rowkeys_qwertz2.xml
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/*
+**
+** Copyright 2012, The Android Open Source Project
+**
+** Licensed under the Apache License, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+**
+**     http://www.apache.org/licenses/LICENSE-2.0
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+*/
+-->
+
+<merge
+    xmlns:latin="http://schemas.android.com/apk/res/com.android.inputmethod.latin"
+>
+    <!-- a,s,d,f,g -->
+    <include
+        latin:keyboardLayout="@xml/rowkeys_qwertz2_left5" />
+    <!-- h,j,k,l -->
+    <include
+        latin:keyboardLayout="@xml/rowkeys_qwertz2_right4" />
+</merge>
diff --git a/java/res/xml/rowkeys_qwertz2_left5.xml b/java/res/xml/rowkeys_qwertz2_left5.xml
new file mode 100644
index 000000000..f0ccaa508
--- /dev/null
+++ b/java/res/xml/rowkeys_qwertz2_left5.xml
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/*
+**
+** Copyright 2014, The Android Open Source Project
+**
+** Licensed under the Apache License, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+**
+**     http://www.apache.org/licenses/LICENSE-2.0
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+*/
+-->
+
+<merge
+    xmlns:latin="http://schemas.android.com/apk/res/com.android.inputmethod.latin"
+>
+    <Key
+        latin:keySpec="a"
+        latin:moreKeys="!text/morekeys_a" />
+    <Key
+        latin:keySpec="s"
+        latin:moreKeys="!text/morekeys_s" />
+    <Key
+        latin:keySpec="d"
+        latin:keyHintLabel="€"
+        latin:additionalMoreKeys="€"
+        latin:moreKeys="!text/morekeys_d" />
+    <Key
+        latin:keySpec="f"
+        latin:keyHintLabel="%"
+        latin:additionalMoreKeys="%" />
+    <Key
+        latin:keySpec="g"
+        latin:keyHintLabel="&amp;"
+        latin:additionalMoreKeys="&amp;"
+        latin:moreKeys="!text/morekeys_g" />
+</merge>
diff --git a/java/res/xml/rowkeys_qwertz2_right4.xml b/java/res/xml/rowkeys_qwertz2_right4.xml
new file mode 100644
index 000000000..c936280d7
--- /dev/null
+++ b/java/res/xml/rowkeys_qwertz2_right4.xml
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/*
+**
+** Copyright 2014, The Android Open Source Project
+**
+** Licensed under the Apache License, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+**
+**     http://www.apache.org/licenses/LICENSE-2.0
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+*/
+-->
+
+<merge
+    xmlns:latin="http://schemas.android.com/apk/res/com.android.inputmethod.latin"
+>
+    <Key
+        latin:keySpec="h"
+        latin:keyHintLabel="-"
+        latin:additionalMoreKeys="-"
+        latin:moreKeys="!text/morekeys_h" />
+    <Key
+        latin:keySpec="j"
+        latin:keyHintLabel="+"
+        latin:additionalMoreKeys="+"
+        latin:moreKeys="!text/morekeys_j" />
+    <Key
+        latin:keySpec="k"
+        latin:keyHintLabel="("
+        latin:additionalMoreKeys="("
+        latin:moreKeys="!text/morekeys_k" />
+    <Key
+        latin:keySpec="l"
+        latin:keyHintLabel=")"
+        latin:additionalMoreKeys=")"
+        latin:moreKeys="!text/morekeys_l" />
+</merge>
diff --git a/java/res/xml/rowkeys_qwertz3.xml b/java/res/xml/rowkeys_qwertz3.xml
index a66c34868..18c60b3a6 100644
--- a/java/res/xml/rowkeys_qwertz3.xml
+++ b/java/res/xml/rowkeys_qwertz3.xml
@@ -23,20 +23,34 @@
 >
     <Key
         latin:keySpec="y"
+        latin:keyHintLabel="*"
+        latin:additionalMoreKeys="*"
         latin:moreKeys="!text/morekeys_y" />
     <Key
-        latin:keySpec="x" />
+        latin:keySpec="x"
+        latin:keyHintLabel="&quot;"
+        latin:additionalMoreKeys="&quot;" />
     <Key
         latin:keySpec="c"
+        latin:keyHintLabel="&apos;"
+        latin:additionalMoreKeys="&apos;"
         latin:moreKeys="!text/morekeys_c" />
     <Key
         latin:keySpec="v"
+        latin:keyHintLabel=":"
+        latin:additionalMoreKeys=":"
         latin:moreKeys="!text/morekeys_v" />
     <Key
-        latin:keySpec="b" />
+        latin:keySpec="b"
+        latin:keyHintLabel=";"
+        latin:additionalMoreKeys=";" />
     <Key
         latin:keySpec="n"
+        latin:keyHintLabel="!"
+        latin:additionalMoreKeys="!"
         latin:moreKeys="!text/morekeys_n" />
     <Key
-        latin:keySpec="m" />
+        latin:keySpec="m"
+        latin:keyHintLabel="\?"
+        latin:additionalMoreKeys="\\?" />
 </merge>
diff --git a/java/res/xml/rows_qwertz.xml b/java/res/xml/rows_qwertz.xml
index 31a147c12..e4ba99fbe 100644
--- a/java/res/xml/rows_qwertz.xml
+++ b/java/res/xml/rows_qwertz.xml
@@ -33,7 +33,7 @@
         latin:keyWidth="10%p"
     >
         <include
-            latin:keyboardLayout="@xml/rowkeys_qwerty2"
+            latin:keyboardLayout="@xml/rowkeys_qwertz2"
             latin:keyXPos="5%p" />
     </Row>
     <Row
-- 
2.17.1

